### PR TITLE
Added personal notes for motions

### PR DESCRIPTION
**Pull request by normanjaeckel**
Missing: Filter personal notes so that every user gets only his own notes in the end.

@FinnStutzenstein You may add for client side stuff first so that we can test this a little bit better.

### CODE REVIEW

**Comment by normanjaeckel** [comment]
@ostcar Can you help me fixing travis? The problem is that retrieving the list of motions I have to fetch data from an uncached new table (personal notes) and therefor fetch data from the user table. If OpenSlides has 100 motions, retrieving all motions costs about 300 queries (but only for the first user, all other users get everything from the cache including my new personal vote data). How can I used personal notes data in motion/serializer.py without making the table a RootRESTModel?

**Comment by normanjaeckel** [comment]
@ostcar Sorry for noise. I found the solution: `GenericRelation`.

**Comment by FinnStutzenstein** [comment]
@normanjaeckel Please check [FinnStutzenstein:PersonalNotesClient](https://github.com/FinnStutzenstein/OpenSlides/tree/PersonalNotesClient). I was testing the client-server interface by trying to send a simple request (go into a detail view, put something in the ugly input field and hit ok). I get multiple server errors:
```
Internal Server Error: /rest/motions/motion/1/set_personal_note/
Traceback (most recent call last):
  File "/home/finn/OpenSlides/.venv/lib/python3.5/site-packages/django/db/models/query.py", line 488, in update_or_create
    obj = self.get(**lookup)
  File "/home/finn/OpenSlides/.venv/lib/python3.5/site-packages/django/db/models/query.py", line 385, in get
    self.model._meta.object_name
openslides.users.models.DoesNotExist: PersonalNote matching query does not exist.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/finn/OpenSlides/.venv/lib/python3.5/site-packages/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
  File "/home/finn/OpenSlides/.venv/lib/python3.5/site-packages/django/db/backends/sqlite3/base.py", line 337, in execute
    return Database.Cursor.execute(self, query, params)
sqlite3.IntegrityError: NOT NULL constraint failed: users_personalnote.user_id

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/finn/OpenSlides/.venv/lib/python3.5/site-packages/django/core/handlers/exception.py", line 39, in inner
    response = get_response(request)
  File "/home/finn/OpenSlides/.venv/lib/python3.5/site-packages/django/core/handlers/base.py", line 187, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/home/finn/OpenSlides/.venv/lib/python3.5/site-packages/channels/handler.py", line 237, in process_exception_by_middleware
    return super(AsgiHandler, self).process_exception_by_middleware(exception, request)
  File "/home/finn/OpenSlides/.venv/lib/python3.5/site-packages/django/core/handlers/base.py", line 185, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/home/finn/OpenSlides/.venv/lib/python3.5/site-packages/django/views/decorators/csrf.py", line 58, in wrapped_view
    return view_func(*args, **kwargs)
  File "/home/finn/OpenSlides/.venv/lib/python3.5/site-packages/rest_framework/viewsets.py", line 87, in view
    return self.dispatch(request, *args, **kwargs)
  File "/home/finn/OpenSlides/.venv/lib/python3.5/site-packages/rest_framework/views.py", line 474, in dispatch
    response = self.handle_exception(exc)
  File "/home/finn/OpenSlides/.venv/lib/python3.5/site-packages/rest_framework/views.py", line 434, in handle_exception
    self.raise_uncaught_exception(exc)
  File "/home/finn/OpenSlides/.venv/lib/python3.5/site-packages/rest_framework/views.py", line 471, in dispatch
    response = handler(request, *args, **kwargs)
  File "/home/finn/OpenSlides/OpenSlides/openslides/motions/views.py", line 368, in set_personal_note
    motion.set_personal_note(request.user, request.data.get('note'), request.data.get('star'))
  File "/home/finn/OpenSlides/OpenSlides/openslides/motions/models.py", line 649, in set_personal_note
    user.set_personal_note(self, note, star)
  File "/home/finn/OpenSlides/OpenSlides/openslides/users/models.py", line 261, in set_personal_note
    defaults=changes,
  File "/home/finn/OpenSlides/.venv/lib/python3.5/site-packages/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/home/finn/OpenSlides/.venv/lib/python3.5/site-packages/django/db/models/query.py", line 490, in update_or_create
    obj, created = self._create_object_from_params(lookup, params)
  File "/home/finn/OpenSlides/.venv/lib/python3.5/site-packages/django/db/models/query.py", line 513, in _create_object_from_params
    six.reraise(*exc_info)
  File "/home/finn/OpenSlides/.venv/lib/python3.5/site-packages/django/utils/six.py", line 686, in reraise
    raise value
  File "/home/finn/OpenSlides/.venv/lib/python3.5/site-packages/django/db/models/query.py", line 505, in _create_object_from_params
    obj = self.create(**params)
  File "/home/finn/OpenSlides/.venv/lib/python3.5/site-packages/django/db/models/query.py", line 399, in create
    obj.save(force_insert=True, using=self.db)
  File "/home/finn/OpenSlides/.venv/lib/python3.5/site-packages/django/db/models/base.py", line 796, in save
    force_update=force_update, update_fields=update_fields)
  File "/home/finn/OpenSlides/.venv/lib/python3.5/site-packages/django/db/models/base.py", line 824, in save_base
    updated = self._save_table(raw, cls, force_insert, force_update, using, update_fields)
  File "/home/finn/OpenSlides/.venv/lib/python3.5/site-packages/django/db/models/base.py", line 908, in _save_table
    result = self._do_insert(cls._base_manager, using, fields, update_pk, raw)
  File "/home/finn/OpenSlides/.venv/lib/python3.5/site-packages/django/db/models/base.py", line 947, in _do_insert
    using=using, raw=raw)
  File "/home/finn/OpenSlides/.venv/lib/python3.5/site-packages/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/home/finn/OpenSlides/.venv/lib/python3.5/site-packages/django/db/models/query.py", line 1045, in _insert
    return query.get_compiler(using=using).execute_sql(return_id)
  File "/home/finn/OpenSlides/.venv/lib/python3.5/site-packages/django/db/models/sql/compiler.py", line 1054, in execute_sql
    cursor.execute(sql, params)
  File "/home/finn/OpenSlides/.venv/lib/python3.5/site-packages/django/db/backends/utils.py", line 79, in execute
    return super(CursorDebugWrapper, self).execute(sql, params)
  File "/home/finn/OpenSlides/.venv/lib/python3.5/site-packages/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
  File "/home/finn/OpenSlides/.venv/lib/python3.5/site-packages/django/db/utils.py", line 94, in __exit__
    six.reraise(dj_exc_type, dj_exc_value, traceback)
  File "/home/finn/OpenSlides/.venv/lib/python3.5/site-packages/django/utils/six.py", line 685, in reraise
    raise value.with_traceback(tb)
  File "/home/finn/OpenSlides/.venv/lib/python3.5/site-packages/django/db/backends/utils.py", line 64, in execute
    return self.cursor.execute(sql, params)
  File "/home/finn/OpenSlides/.venv/lib/python3.5/site-packages/django/db/backends/sqlite3/base.py", line 337, in execute
    return Database.Cursor.execute(self, query, params)
django.db.utils.IntegrityError: NOT NULL constraint failed: users_personalnote.user_id
```

Also one thing I noticed. Nowhere the type of note and star is checked. This has to be added. Currently I can do something like `star: 'fuu'` and the server will crash.

**Comment by FinnStutzenstein** [comment]
@normanjaeckel Sorry, simple fix with `user_id=self.id` in user/models.py at update_or_create.

**Comment by FinnStutzenstein** [comment]
Another notice: The note is bind tusted in the dom. we need a validate_html here.

**Comment by emanuelschuetze** [comment]
@FinnStutzenstein I tested your branch. Client side works and looks good.

@normanjaeckel For server side: Each user should only get its own personal notes. Currently a user can see all personal notes and stars of all users via rest api. That's a serious  privacy issue!

**Comment by normanjaeckel** [comment]
I fixed users/models.py and added a simple bool() check for `star`. I also rebased everything. The privacy fix is done in the next step.

**Comment by normanjaeckel** [comment]
Privacy fix is alos done. @FinnStutzenstein Please rebase your commits on top of this branch and test again.

**Comment by FinnStutzenstein** [comment]
@normanjaeckel Works well, but there is an autoupdate problem.
personalnote
I have two users logged in, both with the same motion. Set the motion as favorite with both users.

If I click on the left window on the filled star, this motions isn't a favorite anymore --> OK.
The right window also changes; no star or note here --> Wrong.

Whats happen is that a autoupdate is send and the left window gets the expected data, but the `personal_notes` for the right window is empty. If I look in the rest api, there is the personal note still there, so it does not get deleted, but the autoupdate does not deliver it.

**Comment by normanjaeckel** [comment]
Please rebase your client commits on top of my (rebased) branch. Then I can test it.

File: openslides/motions/static/js/motions/site.js [hunk 1]
```diff
@@ -1419,6 +1452,19 @@ angular.module('OpenSlidesApp.motions.site', [
...
+            if ($scope.motion.personalNote) {
+                $scope.motion.personalNote.star = !$scope.motion.personalNote.star;
+            } else {
+                $scope.motion.personalNote = {star: true};
+            }
+            $http.put('/rest/motions/motion/' + $scope.motion.id + '/set_personal_note/',
+                $scope.motion.personalNote
+            );
+        };
+        $scope.changePN = function () {
```
Comment (FinnStutzenstein) [inline openslides/motions/static/js/motions/site.js]: This has to be removed.

**Comment by FinnStutzenstein** [comment]
@normanjaeckel Ping. I cannot push into your personalNote branch; now, emanuel did this for me.

File: openslides/motions/static/js/motions/site.js (hunk 1 above)
Comment (normanjaeckel) [inline openslides/motions/static/js/motions/site.js]: Done.

**Comment by normanjaeckel** [comment]
@emanuelschuetze Ready for final test.

**Comment by FinnStutzenstein** [comment]
@emanuelschuetze Should be fixed in my branch

**Comment by normanjaeckel** [comment]
Ready to merge.

**Comment by FinnStutzenstein** [comment]
@emanuelschuetze Please retest again

**Comment by FinnStutzenstein** [comment]
@emanuelschuetze Done.

**Comment by FinnStutzenstein** [comment]
@normanjaeckel @emanuelschuetze Rebased. Please check again, but it should be finished now.

**Comment by emanuelschuetze** [comment]
Successfully retested. Looks good now.

**Pull request merged**